streams/src/main/java/org/apache/kafka/streams/kstream/internals/CogroupedStreamAggregateBuilder.java

```diff
@@ -61,6 +65,24 @@
         groupedStreams.remove(kGrouped);
         kGrouped.ensureCopartitionWith(groupedStreams);
 
+        final Map<KGroupedStreamImpl<K, ?>, StreamsGraphNode> parentNodes = new LinkedHashMap<>();
+
+        for (final KGroupedStreamImpl<K, ?> repartitionReqs : groupPatterns.keySet()) {
+            parentNodes.put(repartitionReqs, repartitionReqs.streamsGraphNode);
```

Comment:
nit: Should this go into an `else` block of the following `if`? Might be easier to read.